packages/core/micro/src/map/producer.ts

```diff
@@ -6,28 +11,28 @@ export abstract class Producer<TMap> implements IProducer<TMap>, IReader<TMap> {
 
     //#region IProducer
 
-    open(consumer: IConsumer<TMap>): IReader<TMap> {
+    public open(consumer: IConsumer<TMap>): IReader<TMap> {
         this.consumers = addConsumer(this.consumers, consumer);
         return this;
     }
 
-    close(consumer: IConsumer<TMap>): void {
+    public close(consumer: IConsumer<TMap>): void {
         this.consumers = removeConsumer(this.consumers, consumer);
     }
 
     //#endregion IProducer
 
     //#region IReader
 
-    public abstract get<K extends keyof TMap>(property: K): TMap[K];
+    public abstract get<K extends keyof TMap>(key: K): TMap[K];
 
-    public get producer() { return this; }
+    public get producer():IProducer<TMap> { return this; }
 
     //#endregion IReader
 
-    protected invalidateValue<K extends keyof TMap>(property: K): void {
+    protected invalidateValue<K extends keyof TMap>(key: K): void {
         forEachConsumer(this.consumers, (consumer) => {
-            consumer.valueChanged(property, this);
+            consumer.keyChanged(key, this);
```

Comment:
The 'key' terminology works well for records & sets, both when observing the "shape" and the "values":

IConsumer {
    // Value associated with the key has changed
    keyChanged(key: string, producer: IShapeProducer
}

IShapeConsumer {
    // The key has been added or removed
    keyChanged(key: string, producer: IShapeProducer
}